crochet-stitcher/src/stages/1-parser.ts

```diff
@@ -1,5 +1,277 @@
-import { ParsedStitch, Pattern } from '../types.js';
+import { Colour, Foundation, ParsedStitch, Pattern, StitchType } from '../types.js';
+
+enum TokenType {
+    Single = 'sc',
+    Double = 'dc',
+    Slip = 'sl',
+    Chain = 'ch',
+    Triple = 'tr',
+    Increment = 'inc',
+    Decrement = 'dec',
+    Invert = 'inv',
+    Comma = ',',
+    LParen = '(',
+    RParen = ')',
+    Number = 'number',
+    Colon = ':',
+    Asterisk = '*',
+    Colour = 'white',
```

Comment:
This looks a bit weird, although I guess there isn't an obvious alternative...

I wonder if the values could be the same as the keys, like enums usually are: `Single = 'Single',` etc?

And, I wonder if the different keywords need separate enum types? A keyword `ParsedToken` could perhaps be represented by `{ type: TokenType.Keyword, value: 'sc' }`